x/escrow/keeper/keeper.go

```diff
@@ -530,18 +562,19 @@ func accountSettleFullblocks(
 	for idx := range payments {
 		p := payments[idx]
 		payments[idx].Balance = p.Balance.Add(
-			sdk.NewCoin(p.Rate.Denom, p.Rate.Amount.Mul(numFullBlocks)))
+			sdk.NewDecCoinFromDec(p.Rate.Denom, p.Rate.Amount.Mul(sdk.NewDecFromInt(numFullBlocks))),
+		)
 	}
 
-	transferred := sdk.NewCoin(blockRate.Denom, blockRate.Amount.Mul(numFullBlocks))
+	transferred := sdk.NewDecCoinFromDec(blockRate.Denom, blockRate.Amount.Mul(sdk.NewDecFromInt(numFullBlocks)))
 
 	account.Transferred = account.Transferred.Add(transferred)
 	account.Balance = account.Balance.Sub(transferred)
 
 	remaining := account.Balance
 	overdrawn := true
 	if numFullBlocks.Equal(heightDelta) {
-		remaining.Amount = sdk.ZeroInt()
+		remaining.Amount = sdk.ZeroDec()
 		overdrawn = false
```

Comment:
does this still work if we're using `TruncateInt()` when calculating `numFullBlocks`?